src/main/java/uk/nhs/digital/nhsconnect/lab/results/model/edifact/ServiceProvider.java

```diff
@@ -0,0 +1,57 @@
+package uk.nhs.digital.nhsconnect.lab.results.model.edifact;
+
+import lombok.Builder;
+import lombok.Getter;
+import lombok.NonNull;
+import lombok.RequiredArgsConstructor;
+import lombok.Setter;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.message.EdifactValidationException;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.message.Split;
+
+/**
+ * Example: SPR+ORG'
+ */
+@Getter
+@Setter
+@Builder
+@RequiredArgsConstructor
```

Comment:
Out of curiosity, how come `@RequiredArgsConstructor` is used here, but in other segments, `@ AllArgsConstructor` is used?

Reply:
Oh that's a mistake. Should be `@AllArgsConstructor` I think

Reply:
I think `@RequiredArgsConstructor` should be used for classes where the fields are `final` and are not null. I'll make sure it's consistent in the classes being added here.

http://www.javabyexamples.com/delombok-allargsconstructor-noargsconstructor-and-requiredargsconstructor